src/app/services/folder.js

```diff
@@ -562,6 +562,48 @@ module.exports = (Model, App) => {
     if (!res) throw new Error();
   };
 
+  const getSize = async (user, folderId) => {
+    const foldersToCheck = [folderId];
+    let totalSize = 0;
+
+    while (foldersToCheck.length > 0) {
+      const currentFolderId = foldersToCheck.shift();
+
+      // Sum files size from this level
+      const filesSize = await getFilesTotalSizeFromFolder(user.id, currentFolderId);
+      totalSize += filesSize;
+
+      // Add folders from this level to the list
+      const folders = await Model.folder
+        .findAll({
+          attributes: ['id'],
+          raw: true,
+          where: {
+            parent_id: { [Op.eq]: currentFolderId },
+            user_id: { [Op.eq]: user.id },
+          },
+        });
+      folders.forEach(folder => foldersToCheck.push(folder.id));
+    }
+
+    return totalSize;
+  };
+
+  const getFilesTotalSizeFromFolder = async (userId, folderId) => {
+    const result = await Model.file.findAll({
+      attributes: [
+        [sequelize.fn('sum', sequelize.col('size')), 'total']
+      ],
+      raw: true,
+      where: {
+        folderId: { [Op.eq]: folderId },
+        userId: { [Op.eq]: userId },
+      },
+    });
+
+    return result[0].total;
```

Comment:
Does `result` always have one element at least? Does this first element always contain a `total` attribute? Just for ensuring that this function is fault-tolerant

Reply:
yes it has always that structure